crates/ra_mbe/src/lib.rs

```diff
@@ -72,10 +106,17 @@ impl MacroRules {
             validate(&rule.lhs)?;
         }
 
-        Ok(MacroRules { rules })
+        // Note that TokenId is started from zero,
+        // We have to add 1 to prevent duplication.
+        let shift = max_id(tt).unwrap_or(0) + 1;
```

Comment:
```suggestion
        let shift = max_id(tt).map_or(0, |it| it + 1);
```

No ids should result in a zero shift

r=me with this change applied